extensions/AbuseFilter/api/ApiQueryAbuseFilters.php

```diff
@@ -75,21 +75,22 @@ public function execute() {
 		$this->addWhereRange( 'af_id', $params['dir'], $params['startid'], $params['endid'] );
 
 		if ( !is_null( $params['show'] ) ) {
-			$show = array_flip( $params['show'] );
+			$show = array_flip( $params[ 'show' ] );
 
 			/* Check for conflicting parameters. */
-			if ( ( isset( $show['enabled'] ) && isset( $show['!enabled'] ) )
-					|| ( isset( $show['deleted'] ) && isset( $show['!deleted'] ) )
-					|| ( isset( $show['private'] ) && isset( $show['!private'] ) ) ) {
-					$this->dieUsage( 'Incorrect parameter - mutually exclusive values may not be supplied', 'show' );
+			if( ( isset( $show[ 'enabled' ] ) && isset( $show[ '!enabled' ] ) )
+				|| ( isset( $show[ 'deleted' ] ) && isset( $show[ '!deleted' ] ) )
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "deleted" 4 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "!deleted" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)